cabot/cabotapp/tasks.py

```diff
@@ -89,11 +89,11 @@ def clean_db(days_to_retain=60):
     from .models import StatusCheckResult, ServiceStatusSnapshot
     from datetime import timedelta
 
-    to_discard_results = StatusCheckResult.objects.filter(time__lte=timezone.now()-timedelta(days=days_to_retain))
-    to_discard_snapshots = ServiceStatusSnapshot.objects.filter(time__lte=timezone.now()-timedelta(days=days_to_retain))
+    to_discard_results = StatusCheckResult.objects.filter(time_complete__lte=timezone.now() - timedelta(days=days_to_retain))
+    to_discard_snapshots = ServiceStatusSnapshot.objects.order_by('time').filter(time__lte=timezone.now() - timedelta(days=days_to_retain))
 
-    result_ids = to_discard_results.values_list('id', flat=True)[:100]
-    snapshot_ids = to_discard_snapshots.values_list('id', flat=True)[:100]
+    result_ids = to_discard_results[:batch_size].values_list('id', flat=True)
```

Comment:
Looks like you only get the count back from delete in django >= 1.9
However the re-fetching with all the id's seems like a massive waste, going to change it to use .count() instead of using len on the id list

Also, I think there's a chance it could constantly be called every 3 seconds if you consistently have at least 1 result every 3 seconds, so going to refactor a little